tsdb/engine/tsm1/file_store.go

```diff
@@ -978,8 +978,20 @@ func (f *FileStore) CreateSnapshot() (string, error) {
 	return tmpPath, nil
 }
 
-// ParseTSMFileName parses the generation and sequence from a TSM file name.
-func ParseTSMFileName(name string) (int, int, error) {
+// FormatFileNameFunc is executed when generating a new TSM filename.
+// Source filenames are provided via src.
+type FormatFileNameFunc func(generation, sequence int, src []string) string
```

Comment:
I think this is adding too much flexibility.  The generation/sequence have specific semantics that apply for compactions and query processing.  I think it would safer to allow the generation to part of the file to be injected.

The contract is that it has to create an increasing integer and cant return an a value less than anything that already exists as a TSM file.